File: Leetcode/containsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) 
    {
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); ++i) 
        {
            if (m.find(nums[i]) != m.end()) return true;
            ++m[nums[i]];
        }
        return false;
    }
};

File: Leetcode/Contains-Duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        container = {}
        for x in nums:
            if x in container:
                container[x] += 1
                if container[x] > 1:
                    return True
            else:
                container[x] = 1
        return False
